js/main.js

```diff
@@ -0,0 +1,27 @@
+var $photoUrl = document.getElementById('photo-url');
+var $entryForm = document.querySelector('.entry-form-container');
+var $entryImage = document.getElementById('entry-image');
+
+$photoUrl.addEventListener('input', function (e) {
+  if ($entryImage.src === '') {
+    $entryImage.style.display = 'none';
+  } else {
+    $entryImage.src = e.target.value;
+  }
+});
+
+$entryForm.addEventListener('submit', function (e) {
+  e.preventDefault();
+  var $entryTitle = document.getElementById('entry-title');
+  var $entryNotes = document.getElementById('entry-notes');
+  var newData = {
+    entryId: data.nextEntryId,
+    title: $entryTitle.value,
+    photoUrl: $photoUrl.value,
+    notesText: $entryNotes.value
+  };
```

Comment:
while this works, I would recommend checking out the `js forms` exercise so that you can take the data directly from the `elements` property of the form. The reason while that would be a better solution is that you would not have to query the DOM for the inputs every time the form is submitted.

Reply:
Thanks. Updated per examples in the js forms exercise.